### PR TITLE
Budget App

### PR DESCRIPTION
**Pull request by Emmy-github-webdev**
## Budget App

In this project I
- Implement authentication
- Implement authorization
- Design category page
- Transaction page
- Add a new category page
- Add a new transaction page
- Write RSpec for the methods

### CODE REVIEW

File: app/views/categories/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<%= form_with(model: category) do |form| %>
+  <div class="category-form">
+    <div class="field">
+      <%= form.text_field :name, placeholder: "Name", class: 'category-input', required: true %>
+    </div>
+
+    <div class="field">
+      <%= form.text_field :icon, placeholder: "Image URL", class: "category-input", required: true %>
+    </div>
+
```
Comment (uwadonat) [inline app/views/categories/_form.html.erb:10]: - [ ] Kindly validate this field to accept only valid URL. when I enter random text like `dss` the application will be broken.
Screenshot from 2022-03-25 14-37-15

**Review by uwadonat — CHANGES_REQUESTED** [review]
Hi **[Emmy-github-webdev](https://github.com/Emmy-github-webdev),** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR  message is very descriptive ✔️
- You did a good presentation ✔️

## Required Changes ♻️

- [ ] Make sure your online version is working properly. Now sign up is not working

- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/categories/_form.html.erb (hunk 1 above)
Comment (Emmy-github-webdev) [reply app/views/categories/_form.html.erb:10]: Thank you @uwadona. Validation added

File: app/models/category.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@ class Category < ApplicationRecord
   belongs_to :user, dependent: :destroy
   has_many :activities_categories
   has_and_belongs_to_many :activities
-  validates_presence_of :name, :icon
+  validates_presence_of :name
+  validates :icon, url: true
 
   def total_amount
     activities.sum(:amount)
```
Comment (Emmyn5600) [inline app/models/category.rb]: - [ ] After adding the image URL, I am getting the following error, kindly fix it to make it run as well as use your project.

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @Emmy-github-webdev 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good PR description :heavy_check_mark:

## Required Changes :recycle:

- [x]  Check the comment under the review.
- [x]  All the tests are not passing successfully
image

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._


image

File: app/models/category.rb (hunk 2 above)
Comment (Emmy-github-webdev) [reply app/models/category.rb]: fixed. Thank you

**Review by descholar-ceo — APPROVED** [review]
Hi @Emmy-github-webdev , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Emmy-github-webdev** [comment]
Thanks @descholar-ceo

**Pull request merged**